### PR TITLE
Bad bad bad

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: src/projection.py [hunk 1]
```diff
@@ -35,44 +35,68 @@ def baseline(img):
...
+letters = [(letter.absolute(), cv2.threshold(cv2.cvtColor(load_image(str(letter.absolute())), cv2.COLOR_BGR2GRAY), 100, 255, cv2.THRESH_BINARY | cv2.THRESH_OTSU)[1])
            for letter in letters]
 # for (letter, img) in letters:
 #     print(letter.name)
 #     img = cv2.bitwise_not(img)
 #     cv2.imwrite(letter.absolute().name, img)
 
 # img = cv2.imread("/home/ramez/PycharmProjects/GradProject/Screenshot_20221212_105958.png", cv2.IMREAD_COLOR)
-img = cv2.imread("/home/ramez/PycharmProjects/GradProject/IMG-20221212-WA0012.jpg", cv2.IMREAD_COLOR)
+img = cv2.imread("C:\\Users\\PC\\Desktop\\ramez\\ArabicOCR\\tifa.png", cv2.IMREAD_COLOR)
```
Comment (ramezgerges) [inline src/projection.py:55]: 5aleeha relative path badal 7war windows w posix da. 5aleeha ./tifa.png

File: src/projection.py [hunk 2]
```diff
@@ -35,44 +35,68 @@ def baseline(img):
...
 
+def load_image(path):
+    stream = open(path, "rb")
+    bytes = bytearray(stream.read())
+    numpyarray = np.asarray(bytes, dtype=np.uint8)
+    return cv2.imdecode(numpyarray, cv2.IMREAD_COLOR)
 
-letters_dir = Path("/home/ramez/PycharmProjects/GradProject/letters")
+
+letters_dir = Path("C:\\Users\\PC\\Desktop\\testop\\ConsoleApp4\\ConsoleApp4\\bin\\Debug\\net7.0-windows\\Images")
```
Comment (ramezgerges) [inline src/projection.py:45]: 5aleeha hena relative path bardo. w 5aly el images f root el repo badal el path el 3ageeb da

File: src/projection.py [hunk 3]
```diff
@@ -35,44 +35,68 @@ def baseline(img):
...
-VP = np.sum(binary_img, 0).astype('int32')  # vertical projection
+# VP = np.sum(binary_img, 0).astype('int32')  # vertical projection
 
 # y2 = normalized(VP, 0)
 # heights = MAX_PLOT_HEIGHT * y2
 # plt.plot(y)
 # plt.show()
 
-d2 = cv2.matchShapes(binary_img, binary_img[:, 350:], cv2.CONTOURS_MATCH_I2, 0)
+# d2 = cv2.matchShapes(binary_img, binary_img[:, 350:], cv2.CONTOURS_MATCH_I2, 0)
```
Comment (ramezgerges) [inline src/projection.py:72]: feel free to remove this

File: src/projection.py [hunk 4]
```diff
@@ -35,44 +35,68 @@ def baseline(img):
...
 # cv2.imshow("lol", binary_img[:][350:])
 # cut = binary_img[:, 384:400]
 # cut = binary_img[:, 312:328]
 # cut = binary_img[:, 350:384]
 # cut = binary_img[:, 328:350]
-cut = binary_img[:, 290:312]
-cv2.imwrite("cut.png", cut)
-for letter, img in letters:
-    print(str(letter), cv2.matchShapes(cut, img, cv2.CONTOURS_MATCH_I2, 0))
```
Comment (ramezgerges) [inline src/projection.py:75]: 5aly dol 3a4an nedebug beehom be2dena

Comment (ramezgerges) [reply src/projection.py:75]: 5aleehom commented bas asdy

**Review by ramezgerges — CHANGES_REQUESTED** [review]

File: src/projection.py (hunk 1 above)
Comment (Youssef1313) [reply src/projection.py:55]:
```suggestion
img = cv2.imread("tifa.png", cv2.IMREAD_COLOR)
```

File: src/projection.py (hunk 2 above)
Comment (ramezgerges) [reply src/projection.py:45]: w 5aleeh unix f gamee3 el 7alat. keda keda Path betzabat el denya

File: src/projection.py (hunk 3 above)
Comment (Youssef1313) [reply src/projection.py:72]:
```suggestion
```

File: src/projection.py [hunk 5]
```diff
@@ -35,44 +35,68 @@ def baseline(img):
...
+
```
Comment (Youssef1313) [inline src/projection.py:78]:
```suggestion
cut = binary_img[:, 290:312]
cv2.imwrite("cut.png", cut)
for letter, img in letters:
    print(str(letter), cv2.matchShapes(cut, img, cv2.CONTOURS_MATCH_I2, 0))

```

Comment (ramezgerges) [reply src/projection.py:81]: 5aleehom commented bel marra ba2a

**Review by ramezgerges — APPROVED** [review]

**Pull request merged**